test.js

```diff
@@ -730,6 +730,230 @@ test('Custom health check', t => {
+  t.plan(9)
+
+  t.test('health check', async t => {
+    t.plan(3)
+    const fastify = Fastify()
+
+    fastify.register(underPressure, {
+      healthCheck: async () => false,
+      healthCheckInterval: 1,
+      pressureHandler: (req, rep, type, value) => {
+        t.equal(type, underPressure.TYPE_HEALTH_CHECK)
+        t.equal(value, undefined)
+        rep.send('B')
+      }
+    })
+
+    fastify.get('/', (req, rep) => rep.send('A'))
+
+    t.equal((await fastify.inject().get('/').end()).body, 'B')
+  })
+
+  t.test('health check - delayed handling with promise success', async t => {
+    t.plan(1)
+    const fastify = Fastify()
+
+    fastify.register(underPressure, {
+      healthCheck: async () => false,
+      healthCheckInterval: 1,
+      pressureHandler: async (req, rep, type, value) => {
+        await wait(250)
+        rep.send('B')
+      }
+    })
+
+    fastify.get('/', (req, rep) => rep.send('A'))
+
+    t.equal((await fastify.inject().get('/').end()).body, 'B')
+  })
+
+  t.test('health check - delayed handling with promise error', async t => {
+    t.plan(2)
+    const fastify = Fastify()
+
+    const errorMessage = 'promiseError'
+
+    fastify.register(underPressure, {
+      healthCheck: async () => false,
+      healthCheckInterval: 1,
+      pressureHandler: async (req, rep, type, value) => {
+        await wait(250)
+        throw new Error(errorMessage)
+      }
+    })
+
+    fastify.get('/', (req, rep) => rep.send('A'))
+
+    const response = await fastify.inject().get('/').end()
+    t.equal(response.statusCode, 500)
+    t.equal(JSON.parse(response.body).message, errorMessage)
+  })
+
+  t.test('health check - no handling', async t => {
+    t.plan(1)
+    const fastify = Fastify()
+
+    fastify.register(underPressure, {
+      healthCheck: async () => false,
+      healthCheckInterval: 1,
+      pressureHandler: (req, rep, type, value) => {}
+    })
+
+    fastify.get('/', (req, rep) => rep.send('A'))
+
+    t.equal((await fastify.inject().get('/').end()).body, 'A')
+  })
+
+  t.test('health check - return response', async t => {
+    t.plan(1)
+    const fastify = Fastify()
+
+    fastify.register(underPressure, {
+      healthCheck: async () => false,
+      healthCheckInterval: 1,
+      pressureHandler: (req, rep, type, value) => 'B'
+    })
+
+    fastify.get('/', (req, rep) => rep.send('A'))
+
+    t.equal((await fastify.inject().get('/').end()).body, 'B')
+  })
+
+  t.test('event loop delay', t => {
+    t.plan(5)
+    const fastify = Fastify()
+
+    fastify.register(underPressure, {
+      maxEventLoopDelay: 1,
+      pressureHandler: (req, rep, type, value) => {
+        t.equal(type, underPressure.TYPE_EVENT_LOOP_DELAY)
+        t.ok(value > 1)
+        rep.send('B')
+      }
+    })
+
+    fastify.get('/', (req, rep) => rep.send('A'))
+
+    fastify.listen(0, async (err, address) => {
+      t.error(err)
+      fastify.server.unref()
+
+      if (monitorEventLoopDelay) {
```

Comment:
I think if this is not available we should skip the test.

Reply:
done but it didnt fix the failing tests completely